pkg/common/aws/s3_session.go

```diff
@@ -0,0 +1,67 @@
+package aws
+
+import (
+	"fmt"
+	"log"
+	"sync"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/credentials"
+	"github.com/aws/aws-sdk-go/aws/session"
+	viper "github.com/openshift/osde2e/pkg/common/concurrentviper"
+	"github.com/openshift/osde2e/pkg/common/config"
+)
+
+const (
+	metricsAWSAccessKeyID     = "metrics.awsAccessKeyId"
+	metricsAWSSecretAccessKey = "metrics.awsSecretAccessKey"
+	metricsAWSRegion          = "metrics.awsRegion"
+
+	metricsAWSAccessKeyIDEnv     = "METRICS_AWS_ACCESS_KEY_ID"
+	metricsAWSSecretAccessKeyEnv = "METRICS_AWS_SECRET_ACCESS_KEY"
+	metricsAWSRegionEnv          = "METRICS_AWS_REGION"
+)
+
+type awsSession struct {
+	session *session.Session
+	once    sync.Once
+}
+
+// AWSSession is the global AWS session for interacting with S3.
+var AWSSession awsSession
```

Comment:
Can this be renamed to `MetricsAWSSession` or something similar? That looks to be all it pertains to

Reply:
I'm okay with nit picky! Commit pushed.